guides/hack/11-built-in-types/56-darray-varray-runtime-options-examples/hack_arr_compat_check_array_key_cast.php

```diff
@@ -0,0 +1,29 @@
+<?hh // strict
+
+namespace Hack\UserDocumentation\Types\VarrayDarrayRuntimeOptions\HackArrCompatCheckArrayKeyCast;
+
+use namespace HHVM\UserDocumentation\_Private;
+
+<<__EntryPoint>>
+async function main_async(): Awaitable<void> {
+  require_once __DIR__.'/../../../../vendor/autoload.hack';
+  \Facebook\AutoloadMap\initialize();
+
+  using _Private\print_short_errors();
+
+  $varray = varray[];
+
+  /*HH_IGNORE_ERROR[4324]*/
+  $varray[1.1] = 'A float?!?';
+  /*HH_IGNORE_ERROR[4324]*/
+  $varray[true] = 'A bool?!?';
+  /*HH_IGNORE_ERROR[4324]*/
+  $varray[null] = 'null?!?';
+
+  $darray = darray[];
+
+  $darray[1.1] = 'A float?!?';
+  $darray[true] = 'A bool?!?';
+  $darray[null] = 'null?!?';
```

Comment:
I find it strange that it is not a typechecker error to `$darray[1.1]` or `$dict[1.1]`.
Varrays, Vectors and vecs **do** have an typechecker error.

Reply:
There's a typechecker flag `disallow_invalid_arraykey` that does this.